data/dataset.py

```diff
@@ -48,23 +48,24 @@ def build_translation_dataset(
 def quantized_piece_to_records(piece: ff.MidiPiece, sequence_len: int, sequence_step: int):
     chopped_sequences = []
     n_samples = 1 + (piece.size - sequence_len) // sequence_step
+    df = piece.df
     for jt in range(n_samples):
         start = jt * sequence_step
         finish = start + sequence_len
-        part = piece[start:finish]
+        part = df.iloc[start:finish]
```

Comment:
Using `MidiPiece` here allows us to track the source exactly, by switching to `df` we make the information about start and finish lost and impossible to track. 

Let me know if this change was to improve performance, if not it would be better to use `piece` here, if yes you can manually add start and finish to the source.